WordPress/gutenberg
# Social Icons: Add a 'Phone' icon which takes a tel: linkadded new phone variation

## What?
This adds a new Phone variation to the Social Link block. It uses the DashIcons phone icon. It has a url attribute of 'tel:' so that users correctly add a phone phone number. 

## Why?
Issue: https://github.com/WordPress/gutenberg/issues/55718
Its a nice addition to the social links block

## How?
This adds a new Phone variation to the Social Link block. It uses the DashIcons phone icon. It has a url attribute of 'tel:' so that users correctly add a phone phone number. 

Do note that because of the URL attribute, this variation is different than all other variations which have an empty URL. If the URL is empty we add an opacity to the icon to indicate to the user that they haven't added a URL yet. We also do not render Social Link Blocks on the front end if they have an empty URL. The phone icon's initial state in non-empty so works a little differently than the others.

Also, the block description from block.json isn't entirely appropriate in this context. ("Display an icon linking to a social media profile or site.") I'm not aware that we can alter that for a variation.

## Testing Instructions
1. Add a Social Icons block to a post or page and add a phone icon to that. 
2. Add a phone number
3. Test the phone icon on the front end. It should trigger a phone call.

### Testing Instructions for Keyboard
1. Add a Social Icons block to a post or page and add a phone icon to that. 
2. Add a phone number
3. Test the phone icon on the front end. It should trigger a phone call.

## Screenshots or screencast 

<img width="490" alt="Screenshot 2023-12-15 at 1 49 36 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/1948493c-a436-420b-9f81-694cfa106b2a">
<img width="299" alt="Screenshot 2023-12-15 at 1 49 43 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/12b19dd5-a6f2-47aa-9201-e7fdcf28e76c">
